db/blob/db_blob_basic_test.cc

```diff
@@ -2011,6 +2011,134 @@ TEST_F(DBBlobWithTimestampTest, MultiGetMergeBlobWithPut) {
   ASSERT_EQ(values[2], "v2_0");
 }
 
+TEST_F(DBBlobWithTimestampTest, IterateBlobs) {
+  Options options = GetDefaultOptions();
+  options.enable_blob_files = true;
+  options.create_if_missing = true;
+  const size_t kTimestampSize = Timestamp(0, 0).size();
+  TestComparator test_cmp(kTimestampSize);
+  options.comparator = &test_cmp;
+
+  DestroyAndReopen(options);
+
+  int num_blobs = 5;
+  std::vector<std::string> keys;
+  std::vector<std::string> blobs;
+
+  WriteOptions write_opts;
+  std::vector<std::string> write_timestamps = {Timestamp(1, 0),
+                                               Timestamp(2, 0)};
+
+  // For each key in ["key0", ... "keyi", ...], write two versions:
+  // Timestamp(1, 0), "blobi0"
+  // Timestamp(2, 0), "blobi1"
+  for (int i = 0; i < num_blobs; i++) {
+    keys.push_back("key" + std::to_string(i));
+    blobs.push_back("blob" + std::to_string(i));
+    for (size_t j = 0; j < write_timestamps.size(); j++) {
+      ASSERT_OK(db_->Put(write_opts, keys[i], write_timestamps[j],
+                         blobs[i] + std::to_string(j)));
+    }
+  }
+  ASSERT_OK(Flush());
+
+  ReadOptions read_options;
+  std::vector<std::string> read_timestamps = {Timestamp(0, 0), Timestamp(3, 0)};
+  Slice ts_lower_bound(read_timestamps[0]);
+  Slice ts_upper_bound(read_timestamps[1]);
+  read_options.iter_start_ts = &ts_lower_bound;
+  read_options.timestamp = &ts_upper_bound;
+
+  auto check_iter_entry = [](const Iterator* iter, std::string expected_key,
+                             std::string expected_ts,
+                             std::string expected_value) {
+    ASSERT_OK(iter->status());
+    std::string expected_ukey_and_ts;
+    expected_ukey_and_ts.assign(expected_key.data(), expected_key.size());
+    expected_ukey_and_ts.append(expected_ts.data(), expected_ts.size());
+
+    ParsedInternalKey parsed_ikey;
+    ASSERT_OK(
+        ParseInternalKey(iter->key(), &parsed_ikey, true /* log_err_key */));
+    ASSERT_EQ(parsed_ikey.user_key, expected_ukey_and_ts);
+    ASSERT_EQ(iter->timestamp().ToString(), expected_ts);
+    ASSERT_EQ(iter->value().ToString(), expected_value);
+  };
```

Comment:
We could save some copies here by passing the `expected_*` parameters by const-ref and comparing `iter->timestamp()` and `expected_ts` as `Slice`s as opposed to `string`s (`Slice` is implicitly constructible from `string`, so `ASSERT_EQ(iter->timestamp(), expected_ts);` should also work). The same goes for `iter->value()` and `expected_value`. Not a big deal though since this is test code.

Reply:
That makes sense. Thank you for the suggestion.